### PR TITLE
Configure Renovate

### PR DESCRIPTION
**Pull request by ms-gh-admin**
Welcome to [Renovate](https://togithub.com/renovatebot/renovate)! This is an onboarding PR to help you understand and configure settings before regular Pull Requests begin.

🚦 To activate Renovate, merge this Pull Request. To disable Renovate, simply close this Pull Request unmerged.

---
### Detected Package Files

 * `docs/requirements.txt` (pip_requirements)
 * `pyproject.toml` (poetry)
### Configuration

🔡 Renovate has detected a custom config for this PR. Feel free to ask for [help](https://togithub.com/renovatebot/renovate/discussions) if you have any doubts and would like it reviewed.

Important: Now that this branch is edited, Renovate can't rebase it from the base branch any more. If you make changes to the base branch that could impact this onboarding PR, please merge them manually.

You have configured Renovate to use branch `develop` as base branch.

### What to Expect

With your current configuration, Renovate will create 10 Pull Requests:

<details>
<summary>Update dependency bandit to v1.7.4</summary>

  - Schedule: ["at any time"]
  - Branch name: `renovate/bandit-1.x-lockfile`
  - Merge into: `develop`
  - Upgrade [bandit](https://togithub.com/PyCQA/bandit) to `1.7.4`

</details>

<details>
<summary>Update dependency coverage to v6.4.2</summary>

  - Schedule: ["at any time"]
  - Branch name: `renovate/coverage-6.x-lockfile`
  - Merge into: `develop`
  - Upgrade [coverage](https://togithub.com/nedbat/coveragepy) to `6.4.2`

</details>

<details>
<summary>Update dependency redis to v4.3.4</summary>

  - Schedule: ["at any time"]
  - Branch name: `renovate/redis-4.x-lockfile`
  - Merge into: `develop`
  - Upgrade [redis](https://togithub.com/redis/redis-py) to `4.3.4`

</details>

<details>
<summary>Update dependency types-toml to v0.10.8</summary>

  - Schedule: ["at any time"]
  - Branch name: `renovate/types-toml-0.x-lockfile`
  - Merge into: `develop`
  - Upgrade [types-toml](https://togithub.com/python/typeshed) to `0.10.8`

</details>

<details>
<summary>Update dependency black to v22.6.0</summary>

  - Schedule: ["at any time"]
  - Branch name: `renovate/black-22.x-lockfile`
  - Merge into: `develop`
  - Upgrade [black](https://togithub.com/psf/black) to `22.6.0`

</details>

<details>
<summary>Update dependency mypy to v0.971</summary>

  - Schedule: ["at any time"]
  - Branch name: `renovate/mypy-0.x-lockfile`
  - Merge into: `develop`
  - Upgrade [mypy](https://togithub.com/python/mypy) to `0.971`

</details>

<details>
<summary>Update dependency pytest to v7.1.2</summary>

  - Schedule: ["at any time"]
  - Branch name: `renovate/pytest-7.x-lockfile`
  - Merge into: `develop`
  - Upgrade [pytest](https://togithub.com/pytest-dev/pytest) to `7.1.2`

</details>

<details>
<summary>Update dependency types-redis to v4.3.7</summary>

  - Schedule: ["at any time"]
  - Branch name: `renovate/types-redis-4.x-lockfile`
  - Merge into: `develop`
  - Upgrade [types-redis](https://togithub.com/python/typeshed) to `4.3.7`

</details>

<details>
<summary>Update dependency yamllint to v1.27.1</summary>

  - Schedule: ["at any time"]
  - Branch name: `renovate/yamllint-1.x-lockfile`
  - Merge into: `develop`
  - Upgrade [yamllint](https://togithub.com/adrienverge/yamllint) to `1.27.1`

</details>

<details>
<summary>Update dependency flake8 to v4</summary>

  - Schedule: ["at any time"]
  - Branch name: `renovate/flake8-4.x-lockfile`
  - Merge into: `develop`
  - Upgrade [flake8](https://togithub.com/pycqa/flake8) to `4.0.1`

</details>

<br />

🚸 Branch creation will be limited to maximum 2 per hour, so it doesn't swamp any CI resources or spam the project. See docs for `prhourlylimit` for details.

---

❓ Got questions? Check out Renovate's [Docs](https://docs.renovatebot.com/), particularly the Getting Started section.
If you need any further assistance then you can also [request help here](https://togithub.com/renovatebot/renovate/discussions).

---

This PR has been generated by [Renovate Bot](https://togithub.com/renovatebot/renovate).

### CODE REVIEW

**Comment by chadell** [comment]
> what Nautobot does for renovate

It makes sense to me. @glennmatthews would make sense to apply this to all the repos?

File: renovate.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      "matchBaseBranches": ["develop"],
+      "schedule": [
+        "before 3am on Friday"
+      ],
+      "major": {"enabled": false},
+      "minor": {"enabled": false}
+    }
+  ],
+  "rangeStrategy": "in-range-only"
+>>>>>>> Stashed changes
```
Comment (chadell) [inline renovate.json:27]:
```suggestion
```

**Review by chadell — APPROVED** [review]
LGTM
question: would make sense to add this in our cookiecutter to keep consistency?

**Comment by Kircheneer** [comment]
> LGTM question: would make sense to add this in our cookiecutter to keep consistency?

I wasn't really 100% done here, but I might just throw it out there like this for now and then iterate upon it similar to what was done in the Nautobot repo. Anyway, I'd like to hold off on contributing this until I have something that I know works well.

**Pull request merged**